### DISCUSSION TITLE
Arduino nano rp2040 connect: ''OSError: [Errno 1] EPERM''

### DISCUSSION THREAD

**Discussion by user5475656** [micropython/micropython#11304] [General]
Hello,

I am using Arduino nano rp2040 connect for a simple WiFi code but having this error: ''OSError: [Errno 1] EPERM'' 

Can you please tell me how to solve this?

Thank you

**Comment by peterhinch** [comment]
You have done better than me. I have wasted hours trying to get the Arduino nano rp2040 connect to work without success. I've just tried again with the latest firmware from the Arduino site, following their instructions; I can't even get a REPL. As far as I'm concerned it's a heap of junk. Even @jimmo had serious problems.

**Reply by robert-hh** [reply]
It's working here. I can connect the AP, run a udptest, ping in an out, run ftp, run a simple tcp data transfer test, ntptime, ...
What I did was updating the Nina firmware to the most recent version. Maybe that's the difference. But I cannot find the instructions on the web site. I have written one for the SAMD port PR, which is here: https://github.com/micropython/micropython/pull/11219/files#diff-4e704793a197f81bc67c2de892ef5bd3297802d66379a300094e9923293e7a17
The NINA is connected to UART 1, the Pin controlling GPIO0 is PIN 2 of the RP2040, Reset is connected to Pin 3.
